StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Link/LinkPaymentController.swift

```diff
@@ -201,20 +230,153 @@ import UIKit
                     }
                     self.payWithLinkContinuation = nil
                 }
         }
     }
+    
+    private func presentWithSDK(
+        from presentingViewController: UIViewController,
+        completionHandler: @escaping (FinancialConnectionsSDKResult?, LinkAccountSession?, NSError?) -> Void
+    ) {
+        let bankAccountCollector = STPBankAccountCollector()
+        
+        let additionalParameters: [String: Any] = [
+            "product": "instant_debits",
+        ]
+        
+        let params = STPCollectBankAccountParams.collectInstantDebitsParams(
+            email: configuration.defaultBillingDetails.email
+        )
+        
+        let elementsSessionContext = makeElementsSessionContext()
+        
+        switch mode {
+        case .paymentIntentClientSecret(let string):
+            bankAccountCollector.collectBankAccountForPayment(
+                clientSecret: string,
+                returnURL: configuration.returnURL,
+                additionalParameters: additionalParameters,
+                elementsSessionContext: elementsSessionContext,
+                onEvent: nil,
+                params: params,
+                from: presentingViewController,
+                financialConnectionsCompletion: completionHandler
+            )
+        case .setupIntentClientSecret(let string):
+            bankAccountCollector.collectBankAccountForSetup(
+                clientSecret: string,
+                returnURL: configuration.returnURL,
+                additionalParameters: additionalParameters,
+                elementsSessionContext: elementsSessionContext,
+                onEvent: nil,
+                params: params,
+                from: presentingViewController,
+                financialConnectionsCompletion: completionHandler
+            )
+        case .deferredIntent(let intentConfiguration):
+            let amount: Int?
+            let currency: String?
+            switch intentConfiguration.mode {
+            case let .payment(amount: _amount, currency: _currency, _, _):
+                amount = _amount
+                currency = _currency
+            case let .setup(currency: _currency, _):
+                amount = nil
+                currency = _currency
+            }
+            bankAccountCollector.collectBankAccountForDeferredIntent(
+                sessionId: "ios_instant_debits_only_\(UUID().uuidString)",
+                returnURL: configuration.returnURL,
+                onEvent: nil,
+                amount: amount,
+                currency: currency,
+                onBehalfOf: nil,
+                additionalParameters: additionalParameters,
+                elementsSessionContext: elementsSessionContext,
+                from: presentingViewController,
+                financialConnectionsCompletion: completionHandler
+            )
+        }
+    }
+    
+    private func makeElementsSessionContext() -> ElementsSessionContext {
```

Comment:
We use the `ElementsSessionContext` to provide prefill details and billing details, which is actually an improvement over the web flow.